File: 1]. DSA + CP/2]. Competitive Programming/05]. GeeksforGeeks/3]. Must Do Coding Questions/01]. Math/1]. Easy/Java/Trailing_Zeros_in_Factorial.java
import java.io.*;
import java.util.*;

class GFG
{
    public static void main(String args[])throws IOException
    {
        Scanner sc = new Scanner(System.in);
        int t = sc.nextInt();
        while(t-- > 0)
        {
            int N = sc.nextInt();
			
            Solution ob = new Solution();
            int ans  = ob.trailingZeroes(N);
            System.out.println(ans);
        }
    }
}

class Solution
{
    static int trailingZeroes(int N)
    {
        // Write your code here
        if(N<0)
            return -1;
        int count=0;
        for(int i=5; N/i>=1; i*=5)
            count += N/i;
        return count;
    }
}

File: 1]. DSA + CP/2]. Competitive Programming/05]. GeeksforGeeks/3]. Must Do Coding Questions/01]. Math/1]. Easy/C++/Trailing_Zeros_in_Factorial.cpp
#include<bits/stdc++.h> 
using namespace std; 

class Solution
{
  public:
    int trailingZeroes(int n)
    {
        // Write Your Code here
        if(n < 0)
            return -1;
        int count = 0;
        for(int i=5; floor(n/i)>=1; i*=5)
            count += floor(n/i);
        return count;
    }
};

int main()
{
    int t;
    cin >> t;
    while (t--)
    {
        int N;
        cin >> N;
        Solution ob;
        int ans  = ob.trailingZeroes(N);
        cout<<ans<<endl;
    }
    return 0;
}
